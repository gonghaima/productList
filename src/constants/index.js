import keyMirror from 'fbjs/lib/keyMirror';

/**
 * @namespace Constants
 * @desc App constants
 */

/**
 * @constant {Object} ActionTypes
 * @memberof Constants
 */
export const ActionTypes = keyMirror({
  SWITCH_MENU: undefined,
  EXCEPTION: undefined,
  PRODUCT_GET_REPOS: undefined,
  PRODUCT_GET_REPOS_SUCCESS: undefined,
  PRODUCT_GET_REPOS_FAILURE: undefined,
  SELECTION_CHANGE: undefined,
  PAGE_CHANGE: undefined,
});

/**
 * @constant {Object} STATUS
 * @memberof Constants
 */
export const STATUS = {
  IDLE: 'idle',
  RUNNING: 'running',
  READY: 'ready',
  SUCCESS: 'success',
  ERROR: 'error',
};
