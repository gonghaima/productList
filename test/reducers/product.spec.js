import reducer from 'reducers/product';
import { ActionTypes } from 'constants/index';

describe('Product', () => {
  it('should return the initial state', () => {
    expect(reducer.product(undefined, {})).toMatchSnapshot();
  });

  it(`should handle ${ActionTypes.PRODUCT_GET_MAKE}`, () => {
    expect(
      reducer.product(undefined, {
        type: ActionTypes.PRODUCT_GET_MAKE,
        payload: { q: 'react' },
      }),
    ).toMatchSnapshot();
  });

  it(`should handle ${ActionTypes.PRODUCT_GET_SUCCESS}`, () => {
    expect(
      reducer.product(undefined, {
        type: ActionTypes.PRODUCT_GET_SUCCESS,
        payload: {},
      }),
    ).toMatchSnapshot();
  });

  it(`should handle ${ActionTypes.PRODUCT_GET_REPOS_FAILURE}`, () => {
    expect(
      reducer.product(undefined, {
        type: ActionTypes.PRODUCT_GET_REPOS_FAILURE,
        payload: {},
      }),
    ).toMatchSnapshot();
  });
});
