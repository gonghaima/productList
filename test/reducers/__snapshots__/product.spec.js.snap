// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Product should handle PRODUCT_GET_MAKE 1`] = `
Object {
  "repos": Object {
    "data": Object {
      "undefined": Array [],
    },
    "message": "",
    "query": undefined,
    "status": "running",
  },
}
`;

exports[`Product should handle PRODUCT_GET_REPOS_FAILURE 1`] = `
Object {
  "repos": Object {
    "data": Object {},
    "message": "Something went wrong",
    "query": "",
    "status": "error",
  },
}
`;

exports[`Product should handle PRODUCT_GET_SUCCESS 1`] = `
Object {
  "repos": Object {
    "data": Object {
      "": Array [],
    },
    "message": "",
    "query": "",
    "status": "ready",
  },
}
`;

exports[`Product should return the initial state 1`] = `
Object {
  "repos": Object {
    "data": Object {},
    "message": "",
    "query": "",
    "status": "idle",
  },
}
`;
